Add metadata state for HCP Packer Registry artifact metadata

nywilken opened the pull request:
Needed by https://github.com/hashicorp/packer/pull/11154

sylviamoss commented on builder/common/artifact.go:
```diff
@@ -66,6 +71,8 @@ func (a *Artifact) State(name string) interface{} {
 	case "atlas.artifact.metadata":
 		return a.stateAtlasMetadata()
+	case "par.artifact.metadata":
+		return a.statePARMetadata()
```
```suggestion
		return a.stateHCPPackerRegistryMetadata()
```

sylviamoss commented on builder/common/artifact.go:
```diff
@@ -119,3 +126,18 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
 	return metadata
 }
+
+func (a *Artifact) statePARMetadata() interface{} {
```
Updating the name like Megan did for the Packer code
```suggestion
// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
// present in this artifact. 
func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
```

sylviamoss commented on builder/common/artifact.go:
```diff
@@ -28,6 +28,11 @@ type Artifact struct {
 	Session *session.Session
 }
 
+type registryImage struct {
```
```suggestion
// hcpPackerRegistryImage provides the necessary metadata information that Packer will push to HCP Packer Registry
type hcpPackerRegistryImage struct {
```

sylviamoss commented on builder/common/artifact.go:
```diff
@@ -119,3 +126,18 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
+	metadata := make([]registryImage, 0, len(a.Amis))
+
+	for region, imageId := range a.Amis {
+		metadata = append(metadata, registryImage{
```
```suggestion
		metadata = append(metadata, hcpPackerRegistryImage{
```

sylviamoss commented on builder/common/artifact_test.go:
```diff
@@ -88,3 +88,28 @@ func TestArtifactState(t *testing.T) {
+	}
+
+	actual := a.State("par.artifact.metadata")
+	expected := []registryImage{
```
```suggestion
	expected := []hcpPackerRegistryImage{
```

sylviamoss commented on builder/common/artifact_test.go:
```diff
@@ -88,3 +88,28 @@ func TestArtifactState(t *testing.T) {
 	}
 }
+
+func TestArtifactState_parMetadata(t *testing.T) {
```
```suggestion
func TestArtifactState_hcpPackerRegistryMetadata(t *testing.T) {
```

sylviamoss reviewed (approved):
This looks good! 👍🏼  🎉  I added some comments and renamed PAR to HCP Packer Registry. 
Feel free to change the comments with a better text :D

nywilken commented on builder/common/artifact.go:
```diff
@@ -119,3 +129,20 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
+// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
+// present in this artifact. 
+func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
+	metadata := make([]registryImage, 0, len(a.Amis))
```
```suggestion
	metadata := make([] hcpPackerRegistryImage, 0, len(a.Amis))
```

nywilken commented:
> This looks good! 👍🏼 🎉 I added some comments and renamed PAR to HCP Packer Registry.
> Feel free to change the comments with a better text :D

Awesome, thanks for the review. The changes look great. I committed and will merge once all is green.

nywilken commented on builder/common/artifact.go:
```diff
@@ -119,3 +129,20 @@ func (a *Artifact) stateAtlasMetadata() interface{} {
+// stateHCPPackerRegistryMetadata will write the metadata as an hcpRegistryImage for each of the AMIs 
+// present in this artifact. 
+func (a *Artifact) stateHCPPackerRegistryMetadata() interface{} {
+	metadata := make([] hcpPackerRegistryImage, 0, len(a.Amis))
```
```suggestion
	metadata := make([]hcpPackerRegistryImage, 0, len(a.Amis))
```
